mesonbuild/compilers/c.py

```diff
@@ -651,14 +651,9 @@ def _have_prototype_templ():
         # This may include, for instance _GNU_SOURCE which must be defined
         # before limits.h, which includes features.h
         head = '{prefix}\n#include <limits.h>\n'
-        # We don't know what the function takes or returns, so return it as an int.
-        # Just taking the address or comparing it to void is not enough because
-        # compilers are smart enough to optimize it away. The resulting binary
-        # is not run so we don't care what the return value is.
+        # We don't know what the function takes or returns
         main = '''\nint main() {{
-            void *a = (void*) &{func};
-            long b = (long) a;
-            return (int) b;
+            {func};
```

Comment:
This change is correct. We only changed this in 82bb24b264a1d46c7037b69f1cc10b123bbcf9b5 because of https://github.com/mesonbuild/meson/issues/1053, which is no longer needed.

Reply:
Note that on IRC it was identified that this seems to break with the Android SDK somehow. I do not have that SDK so I can't easily test that myself. (Apparently just `func;` does not always cause the compiler to emit code that would need the linker to actually link, or something like that, but I did not fully understand how this happened.)